### PR TITLE
Ovl_En_Cha OK

### PR DESCRIPTION
**Pull request by hensldm**

### CODE REVIEW

File: src/overlays/actors/ovl_En_Cha/z_en_cha.c [hunk 1]
```diff
@@ -21,16 +24,98 @@ const ActorInit En_Cha_InitVars = {
...
+    if (this->actor.cutscene == -1) {
+        this->actionFunc = func_80BEB654;
+        return;
+    }
+    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->actor.cutscene, &this->actor);
+        this->actionFunc = func_80BEB654;
+        return;
+    }
+    ActorCutscene_SetIntentToPlay(this->actor.cutscene);
```
Comment (jpburnett) [inline src/overlays/actors/ovl_En_Cha/z_en_cha.c:81]:
```suggestion
    if (this->actor.cutscene == -1) {
        this->actionFunc = func_80BEB654;
    }
    else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
        ActorCutscene_StartAndSetUnkLinkFields(this->actor.cutscene, &this->actor);
        this->actionFunc = func_80BEB654;
    } else {
        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
    }
```
Just out of curiosity, does this work to get rid of the returns?

**Review by jpburnett — COMMENTED** [review]
Looks good, just a couple thoughts:
1. see the suggestion on the return statements
2. It could just be me, and others can chime in, I like having a description up top to know what actor I am looking at since the names aren't always indicative. ex: 
```
/*
 * File: z_en_cha.c
 * Overlay: ovl_En_Cha
 * Description: Laundry Pool Bell
 */
 ```

File: src/overlays/actors/ovl_En_Cha/z_en_cha.c [hunk 2]
```diff
@@ -21,16 +30,97 @@ const ActorInit En_Cha_InitVars = {
...
+    else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->actor.cutscene, &this->actor);
+        this->actionFunc = func_80BEB654;
+    } else {
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Cha_0x80BEB520/EnCha_Destroy.asm")
+void func_80BEB654(EnCha *this, GlobalContext *globalCtx) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_Cha/z_en_cha.c:89]: This looks to be the bell's idle function.
Can you please rename this to something like
`EnCha_Idle`

File: src/overlays/actors/ovl_En_Cha/z_en_cha.c [hunk 3]
```diff
@@ -21,16 +30,97 @@ const ActorInit En_Cha_InitVars = {
...
+    gSaveContext.perm.weekEventReg[60] &= 0xFB;
+}
+
+void EnCha_Destroy(Actor *thisx, GlobalContext *globalCtx) {
+    EnCha *this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80BEB5DC(EnCha *this, GlobalContext *globalCtx) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_Cha/z_en_cha.c:76]: This looks to be the action state that handles the cutscene that brings out Kaffei.
Can you please rename this to something like
`EnCha_Ring`

**Review by Kenix3 — CHANGES_REQUESTED** [review]

**Review by jpburnett — APPROVED** [review]
looks clean 👍

**Review by Kenix3 — APPROVED** [review]
Looks good to me!

**Pull request merged**